apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/_components/Sidebar/PublishDraftCommitModal/index.tsx

```diff
@@ -0,0 +1,60 @@
+import { useMemo } from 'react'
+
+import {
+  ConfirmModal,
+  useCurrentProject,
+  useToast,
+} from '@latitude-data/web-ui'
+import { deleteDraftCommitAction } from '$/actions/commits/deleteDraftCommitAction'
+import { ROUTES } from '$/services/routes'
+import useCommits from '$/stores/commitsStore'
+import { ReactStateDispatch } from '$ui/lib/commonTypes'
+import { useRouter } from 'next/navigation'
+import { useServerAction } from 'zsa-react'
+
+export default function PublishDraftCommitModal({
+  commitId,
+  onClose,
+}: {
+  commitId: number | null
+  onClose: ReactStateDispatch<number | null>
+}) {
+  const { data, mutate } = useCommits()
+  const { toast } = useToast()
+  const commit = useMemo(() => data.find((c) => c.id === commitId), [commitId])
+  const { project } = useCurrentProject()
+  const router = useRouter()
+  const { execute, isPending } = useServerAction(deleteDraftCommitAction, {
```

Comment:
This will be done in another PR. Publishing requires more work. So ignore that this is using delete action